src/pipeline.ts

```diff
@@ -482,14 +523,400 @@ export async function scanRssFeeds() {
+		});
+	}
+	if (!fs.existsSync(targetDir) || !fs.statSync(targetDir).isDirectory()) {
+		logger.info({
+			label: Label.INJECT,
+			message: `No directory found at ${targetDirLog} - ensure it exists and is accessible (verify docker volumes if applicable)`,
+		});
+		return;
+	}
+	const dirContents = await findAllTorrentFilesInDir(targetDir);
+	if (dirContents.length === 0) {
+		logger.info({
+			label: Label.INJECT,
+			message: `No torrent files found to inject in ${targetDirLog}`,
+		});
+		return;
+	}
+	logger.info({
+		label: Label.INJECT,
+		message: `Found ${chalk.bold.white(dirContents.length)} torrent file(s) to inject in ${targetDirLog}`,
+	});
+
+	const { searchees } = await findSearchableTorrents(Label.INJECT, {
+		useFilters: false,
+	});
+
+	const toDelete = new Set<string>();
+	const toRecheck = new Set<string>();
+
+	// Usually source got deleted or partial injection never completes
+	function shouldCleanUpTorrent(torrentFilePath: string) {
+		return fs.statSync(torrentFilePath).mtimeMs < Date.now() - ms("1 week");
+	}
+
+	let totalInjected = 0;
+	let totalFullMatches = 0;
+	let totalPartialMatches = 0;
+	let totalBlocked = 0;
+	let totalAlreadyExists = 0;
+	let totalCandidateIncomplete = 0;
+	let totalSearcheeIncomplete = 0;
+	let totalFailed = 0;
+	let totalUnmatched = 0;
+	let count = 0;
+	let foundBadFormat = false;
+	for (const torrentFilePath of dirContents) {
+		const progress = chalk.blue(`(${++count}/${dirContents.length})`);
+		const filePathLog = chalk.bold.magenta(torrentFilePath);
+		if (shouldCleanUpTorrent(torrentFilePath)) {
+			toDelete.add(torrentFilePath);
+		}
+		let meta: Metafile;
+		try {
+			meta = await parseTorrentFromFilename(torrentFilePath);
+		} catch (e) {
+			logger.error({
+				label: Label.INJECT,
+				message: `${progress} Failed to parse ${filePathLog}`,
+			});
+			logger.debug(e);
+			continue;
+		}
+		const metaLog = getLogString(meta, chalk.bold.white);
+
+		const torrentNameInfo =
+			await parseInfoFromSavedTorrent(torrentFilePath);
+		const tracker = torrentNameInfo?.tracker ?? UNKNOWN_TRACKER;
+		if (tracker === UNKNOWN_TRACKER) {
+			foundBadFormat = true;
+		}
+
+		// Decide stage
+		const fullMatches: [
+			SearcheeWithLabel,
+			Decision.MATCH | Decision.MATCH_SIZE_ONLY,
+		][] = [];
+		const partialMatches: [SearcheeWithLabel, Decision.MATCH_PARTIAL][] =
+			[];
+		let foundBlocked = false;
+		for (const searchee of searchees) {
+			const { decision } = await assessCandidateHelper(
+				meta,
+				searchee,
+				[],
+			);
+			if (decision === Decision.MATCH) {
+				fullMatches.unshift([searchee, decision]);
+			} else if (decision === Decision.MATCH_SIZE_ONLY) {
+				fullMatches.push([searchee, decision]);
+			} else if (decision === Decision.MATCH_PARTIAL) {
+				partialMatches.push([searchee, decision]);
+			} else if (decision === Decision.BLOCKED_RELEASE) {
+				foundBlocked = true;
+			}
+		}
```

Comment:
I'm curious—could we run these through `checkNewCandidateMatch` to figure out what searchee they belong to, rather than checking them against every searchee? Or is there some reason we wouldn't want to do that?

Reply:
We can't use `checkNewCandidateMatch` as we need information on every matched searchee. We also want to check data based searchees as well. We could the reverse lookup process though with some tweaking.

I think checking against every searchee is easiest and most complete. It takes just a couple seconds to process each torrent file as it is.

The only negative with the current implementation is that it could link an unrelated torrent since we are only checking size in risky and partial match modes. I haven't seen any issues with this yet (and it's extremely unlikely), but I'll just verify each successful match with a levenshtein check on the titles.

Reply:
Why do we need information on every matched searchee?